deployer/deployer-services/src/main/java/io/hyscale/deployer/services/deployer/impl/KubernetesDeployer.java

```diff
@@ -267,13 +250,18 @@ public InputStream logs(K8sAuthorisation authConfig, String serviceName, String
     }
 
     @Override
-    public ServiceAddress getServiceAddress(DeploymentContext context) throws HyscaleException {
-        ServiceAddress serviceAddress = null;
+    public ServiceAddress  getServiceAddress(DeploymentContext context) throws HyscaleException {
+        ServiceAddress serviceAddress;
         try {
             ApiClient apiClient = clientProvider.get((K8sAuthorisation) context.getAuthConfig());
+            String selector = ResourceSelectorUtil.getServiceSelector(context.getAppName(), context.getServiceName());
+            PodParent podParent = podParentProvider.getPodParent(apiClient, context.getAppName(), context.getServiceName(), context.getNamespace());
+            LoadBalancer loadBalancer = PodParentUtil.getLoadBalancerInSpec(podParent);
+            if (loadBalancer != null) {
+                return K8sServiceUtil.getLBServiceAddress(loadBalancer, apiClient, selector, context.getNamespace());
```

Comment:
While getting the service address through v1ServiceHandler we also have the option to wait as the IP allocation takes some time. Can that case occur here as well? or is the service address available immediately?